packages/edit-post/src/components/options-modal/index.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import { PostTaxonomies } from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import { EnablePublishSidebarOption, EnableTipsOption, EnablePanelOption } from './options';
+
+const MODAL_NAME = 'edit-post/options';
+
+const Section = ( { title, children } ) => (
+	<section className="edit-post-options__section">
+		<h2 className="edit-post-options__section-title">{ title }</h2>
+		{ children }
+	</section>
+);
+
+export function OptionsModal( { isModalActive, toggleModal } ) {
+	if ( ! isModalActive ) {
+		return null;
+	}
+
+	return (
+		<Modal
+			className="edit-post-options"
+			title={ <span className="edit-post-options__title">{ __( 'Options' ) }</span> }
+			closeLabel={ __( 'Close' ) }
+			onRequestClose={ toggleModal }
+		>
+			<Section title={ __( 'General' ) }>
+				<EnablePublishSidebarOption label={ __( 'Enable Pre-publish Checks' ) } />
+				<EnableTipsOption label={ __( 'Enable Tips' ) } />
+			</Section>
+			<Section title={ __( 'Document Panels' ) }>
+				<EnablePanelOption panelName="post-status" label={ __( 'Status & Visibility' ) } />
+				<PostTaxonomies
```

Comment:
@talldan we can't do much about it, this can change depending on the website when a plugin registers their own taxonomy.

@noisysocks is this something that is already pre-cached by PHP, if not maybe we should consider doing it on PHP side to avoid network roundtrip. We do it already for a few essential API calls:

https://github.com/WordPress/gutenberg/blob/master/lib/client-assets.php#L1303-L1310

It looks like taxonomies is include, so it shouldn't be an issue here. We should double check.

Reply:
You're right that it taxonomies are already included in the list of API calls to cache.

When I test this locally it doesn't seem to be working, but since this issue also exists on `master` I will look it separately.